app/controllers/admin/masquerade_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Admin
+  class MasqueradeController < ::ApplicationController # does not inherit from administrate
```

Comment:
Does it not inherit from Administrate so that it's impossible to masquerade into the Administrate dashboard and change user permissions?

Reply:
Ah good question!  I haven't looked at how this would interact with the administrate routes.  I did this here because I wanted to know exactly what code was running here, and with the administrate `Admin::ApplicationController` there are more moving parts that aren't visible in our codebase (eg, I think it defines `requested_resource` and reflects on the controller class name to know what model to look at, and I wasn't sure how that applied here).